gateway/src/apicast/policy/routing/routing_operation.lua

```diff
@@ -0,0 +1,66 @@
+--- RoutingOperation
+-- This module is based on the Operation one. The only difference is that
+-- operations for the routing policy check request information that is not
+-- available when the operation is instantiated, like headers, query arguments,
+-- etc. That is the reason why in instances of this module, there are functions
+-- to get the left operand instead of the operand itself.
+
+local setmetatable = setmetatable
+local Operation = require('apicast.conditions.operation')
+
+local _M = {}
+
+local mt = { __index = _M }
+
+local function new(evaluate_left_side_func, op, value, value_type)
+  local self = setmetatable({}, mt)
+
+  self.evaluate_left_side_func = evaluate_left_side_func
+  self.op = op
+  self.value = value
+  self.value_type = value_type
+
+  return self
+end
+
+function _M.new_op_with_path(op, value, value_type)
+  local eval_left_func = function(request) return request:get_uri() end
+  return new(eval_left_func, op, value, value_type)
+end
+
+function _M.new_op_with_header(header_name, op, value, value_type)
+  local eval_left_func = function(request)
+    return request:get_header(header_name)
```

Comment:
Allocation function for each operation might be a bit resource intensive, no?
But I guess it might not be that bad when they are allocated only when they are instantiated and not on evaluation.

Reply:
Yes, I don't think this has an impact on perf because it's only executed when instantiated as you said. However, in this case, extracting those functions to locals is an easy change and might make things a bit clearer, so I'll do that :+1:

Reply:
Looks like they depend on the closure, so they'd need to accept additional parameters. And then they become polymorphic interface and can reduce the whole `new_op_with_*` into one constructor and injecting a function :)

Reply:
Oh right, didn't see that it uses `header_name`. I'll leave it as it is, then.